services/search/src/search/routes/search.py

```diff
@@ -0,0 +1,300 @@
+    duckdb_index_file_directory: StrPath,
+    cached_assets_base_url: str,
+    cached_assets_directory: StrPath,
+    target_revision: str,
+    cache_max_days: int,
+    hf_endpoint: str,
+    external_auth_url: Optional[str] = None,
+    hf_token: Optional[str] = None,
+    hf_jwt_public_key: Optional[str] = None,
+    hf_jwt_algorithm: Optional[str] = None,
+    hf_timeout_seconds: Optional[float] = None,
+    max_age_long: int = 0,
+    max_age_short: int = 0,
+) -> Endpoint:
+    async def search_endpoint(request: Request) -> Response:
+        revision: Optional[str] = None
+        with StepProfiler(method="search_endpoint", step="all"):
+            try:
+                with StepProfiler(method="search_endpoint", step="validate parameters"):
+                    dataset = request.query_params.get("dataset")
+                    config = request.query_params.get("config")
+                    split = request.query_params.get("split")
+                    query = request.query_params.get("query")
+
+                    if (
+                        not dataset
+                        or not config
+                        or not split
+                        or not query
+                        or not are_valid_parameters([dataset, config, split, query])
+                    ):
+                        raise MissingRequiredParameterError(
+                            "Parameter 'dataset', 'config', 'split' and 'query' are required"
+                        )
+
+                    offset = int(request.query_params.get("offset", 0))
+                    if offset < 0:
+                        raise InvalidParameterError(message="Offset must be positive")
+
+                    length = int(request.query_params.get("length", MAX_ROWS))
+                    if length < 0:
+                        raise InvalidParameterError("Length must be positive")
+                    if length > MAX_ROWS:
+                        raise InvalidParameterError(f"Length must be less than or equal to {MAX_ROWS}")
+
+                with StepProfiler(method="search_endpoint", step="check authentication"):
+                    # if auth_check fails, it will raise an exception that will be caught below
+                    auth_check(
+                        dataset=dataset,
+                        external_auth_url=external_auth_url,
+                        request=request,
+                        hf_jwt_public_key=hf_jwt_public_key,
+                        hf_jwt_algorithm=hf_jwt_algorithm,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                    )
+
+                logging.info(f"/search {dataset=} {config=} {split=} {query=} {offset=} {length=}")
+
+                with StepProfiler(method="search_endpoint", step="validate indexing was done"):
+                    # no cache data is needed to download the index file
+                    # but will help to validate if indexing was done
```

Comment:
why not use the content of the cached response to get the URL, instead of forging it? It seems less brittle to me

Reply:
Not sure if I can use an URL for hf_hub_download. Doc https://huggingface.co/docs/huggingface_hub/v0.16.3/en/package_reference/file_download#huggingface_hub.hf_hub_download says that basically it is just needed  repo_type, revision, repo_id, filename, local_dir, token. Anyway, I removed DEFAULT_INDEX_NAME in favor of taking the filename from cache response and use it when calling download function.

Reply:
hmmm ok